### PR TITLE
test(theme): `<ThemeProvider/>`

### PR DESCRIPTION
**Pull request by ogonkov**

### CODE REVIEW

File: src/components/Table/hoc/withTableActions/withTableActions.tsx [hunk 1]
```diff
@@ -62,7 +63,7 @@ const BUTTON_CLASSNAME = b('actions-button');
 export function withTableActions<I extends TableDataItem, E extends {} = {}>(
     TableComponent: React.ComponentType<TableProps<I> & E>,
 ): React.ComponentType<TableProps<I> & WithTableActionsProps<I> & E> {
-    const componentName = TableComponent.displayName || TableComponent.name || 'Component';
+    const componentName = getComponentName(TableComponent);
```
Comment (ValeraS) [inline src/components/Table/hoc/withTableActions/withTableActions.tsx:66]: Please extract everything related to `getComponentName` into another PR.

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/components/Table/hoc/withTableActions/withTableActions.tsx (hunk 1 above)
Comment (ogonkov) [reply src/components/Table/hoc/withTableActions/withTableActions.tsx:66]: Done #257

File: src/components/theme/__tests__/ThemeProvider.test.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    render(
+        <ThemeProvider theme="light">
+            <ThemeDisplay />
+        </ThemeProvider>,
+    );
+
+    mock.changeMedia(true);
+
+    const el = await screen.findByText('Theme: light, theme value: light');
+    expect(el).toBeInTheDocument();
```
Comment (ValeraS) [inline src/components/theme/__tests__/ThemeProvider.test.tsx:62]: To say it doesn't change, we also need to check `screen.getByText('Theme: light, theme value: light');` before `mock.changeMedia`, right?

File: test-utils/mocks/mockMatchMedia.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import {act} from '@testing-library/react';
+
+export function mockMatchMedia(matches = false) {
+    const ogMatchMedia = window.matchMedia;
```
Comment (ValeraS) [inline test-utils/mocks/mockMatchMedia.ts:4]: What does `og` mean?

**Review by ValeraS — CHANGES_REQUESTED** [review]
Can we also test body class?

File: src/components/theme/__tests__/ThemeProvider.test.tsx (hunk 2 above)
Comment (ogonkov) [reply src/components/theme/__tests__/ThemeProvider.test.tsx:62]: We already assert it in previous test

Comment (ValeraS) [reply src/components/theme/__tests__/ThemeProvider.test.tsx:62]: We assert it for `system` them

File: test-utils/mocks/mockMatchMedia.ts (hunk 3 above)
Comment (ogonkov) [reply test-utils/mocks/mockMatchMedia.ts:4]: og = original

https://www.merriam-webster.com/dictionary/OG

but i guess it was bad decision for var

**Comment by ogonkov** [comment]
> Can we also test body class?

Assert that body got specific class name? We cant guarantee anything with this test

**Comment by ValeraS** [comment]
> > Can we also test body class?
> 
> Assert that body got specific class name? We cant guarantee anything with this test

Why not? We expect a specific class to be set for the body.

File: src/components/theme/__tests__/ThemeProvider.test.tsx (hunk 2 above)
Comment (ogonkov) [reply src/components/theme/__tests__/ThemeProvider.test.tsx:62]: Fixed

**Pull request closed without merging**